OpenApiOptions fails to initialize without jackson-module-kotlin

**Actual behavior (the bug)**
When you don't have `jackson-module-kotlin` installed, initializing `OpenApiOptions` fails before you can specify a replacement `ToJsonMapper`.

```
Exception in thread "main" java.lang.NoClassDefFoundError: com/fasterxml/jackson/module/kotlin/ExtensionsKt
	at io.javalin.plugin.openapi.jackson.JacksonToJsonMapper.<clinit>(JacksonToJsonMapper.kt:12)
	at io.javalin.plugin.openapi.OpenApiOptions.<init>(OpenApiOptions.kt:42)
	at io.javalin.plugin.openapi.OpenApiOptions.<init>(OpenApiOptions.kt:48)
	at io.javalin.plugin.openapi.OpenApiOptions.<init>(OpenApiOptions.kt:49)
	...........registerSwagger(Server.java:171)
        ..........
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.module.kotlin.ExtensionsKt
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 7 more
```

**Expected behavior**
The `OpenApiOptions` should allow me to specify an alternate `ToJsonMapper` before trying to initialize the default one provided by the missing `jackson-module-kotlin`.

**To Reproduce**
```
private void registerSwagger(Javalin app) {
    final Info applicationInfo = new Info()
                .version("1.0")
                .description("My Application");
    final OpenApiOptions options = new OpenApiOptions(applicationInfo).path("/swagger-docs");

    final Gson gson = new Gson();
    options.setToJsonMapper(gson::toJson);

    app.config.registerPlugin(new OpenApiPlugin(options));
}
```

--- a/src/main/java/io/javalin/plugin/openapi/OpenApiOptions.kt
+++ b/src/main/java/io/javalin/plugin/openapi/OpenApiOptions.kt
@@ -8,6 +8,7 @@ import io.javalin.plugin.openapi.jackson.JacksonModelConverterFactory
 import io.javalin.plugin.openapi.jackson.JacksonToJsonMapper
 import io.javalin.plugin.openapi.ui.ReDocOptions
 import io.javalin.plugin.openapi.ui.SwaggerOptions
+import io.javalin.plugin.openapi.utils.LazyDefaultValue
 import io.swagger.v3.oas.models.OpenAPI
 import io.swagger.v3.oas.models.info.Info
 
@@ -34,12 +35,12 @@ class OpenApiOptions constructor(val createBaseConfiguration: () -> OpenAPI) {
      * Creates a model converter, which converts a class to an open api schema.
      * Defaults to the jackson converter.
      */
-    var modelConverterFactory: ModelConverterFactory = JacksonModelConverterFactory
+    var modelConverterFactory: ModelConverterFactory by LazyDefaultValue { JacksonModelConverterFactory }
     /**
      * The json mapper for creating the object api schema json. This is separated from
      * the default JavalinJson mappers.
      */
-    var toJsonMapper: ToJsonMapper = JacksonToJsonMapper
+    var toJsonMapper: ToJsonMapper by LazyDefaultValue { JacksonToJsonMapper }
     /**
      * A list of package prefixes to scan for annotations.
      */

--- a/src/main/java/io/javalin/plugin/openapi/jackson/JacksonToJsonMapper.kt
+++ b/src/main/java/io/javalin/plugin/openapi/jackson/JacksonToJsonMapper.kt
@@ -9,8 +9,7 @@ import io.javalin.plugin.json.ToJsonMapper
  * This enables some of the options that are required to work with the uis.
  */
 object JacksonToJsonMapper : ToJsonMapper {
-    val objectMapper = jacksonObjectMapper()
-            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    val objectMapper by lazy { jacksonObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL) }
 
     override fun map(obj: Any): String = objectMapper.writeValueAsString(obj)
 }

--- a/src/main/java/io/javalin/plugin/openapi/utils/LazyDefaultValue.kt
+++ b/src/main/java/io/javalin/plugin/openapi/utils/LazyDefaultValue.kt
@@ -1,0 +1,16 @@
+package io.javalin.plugin.openapi.utils
+
+import kotlin.reflect.KProperty
+
+internal class LazyDefaultValue<T>(private val init: () -> T) {
+    var value: T? = null
+
+    operator fun getValue(thisRef: Any?, property: KProperty<*>): T {
+        value = value ?: init()
+        return value!!
+    }
+
+    operator fun setValue(thisRef: Any?, property: KProperty<*>, value: T) {
+        this.value = value
+    }
+}

--- a/src/test/java/io/javalin/openapi/utils/LazyDefaultValueTest.kt
+++ b/src/test/java/io/javalin/openapi/utils/LazyDefaultValueTest.kt
@@ -1,0 +1,43 @@
+package io.javalin.openapi.utils
+
+import io.javalin.plugin.openapi.utils.LazyDefaultValue
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.util.function.Supplier
+
+class LazyDefaultValueTest {
+    @Test
+    fun `should init default value if not overriden`() {
+        val mockSupplier = mockk<Supplier<Int>>()
+        every { mockSupplier.get() } returns 0
+
+        class MyClass {
+            var value by LazyDefaultValue { mockSupplier.get() }
+        }
+
+        val myObject = MyClass()
+
+        assertThat(myObject.value).isEqualTo(0)
+    }
+
+    @Test
+    fun `should not call init function if overriden`() {
+        val mockSupplier = mockk<Supplier<Int>>()
+        every { mockSupplier.get() } returns 0
+
+        class MyClass {
+            var value by LazyDefaultValue { mockSupplier.get() }
+        }
+
+        val myObject = MyClass()
+        myObject.value = 1
+
+        assertThat(myObject.value).isEqualTo(1)
+        verify(exactly = 0) { mockSupplier.get() }
+        confirmVerified(mockSupplier)
+    }
+}

